### PR TITLE
feat: ColorPicker support cssVar

### PR DESCRIPTION
**Pull request by RedJue**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
- #45618

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    ColorPicker support cssVar       |
| 🇨🇳 Chinese |   颜色选择器支持 css 变量        |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 96c7c60</samp>

This pull request refactors the color picker component to use hooks and CSS variables for better compatibility with server-side rendering and design tokens. It also introduces the `@ant-design/cssinjs` package for creating and manipulating CSS objects with units and arithmetic operations. It updates the styles of the component and its subcomponents to improve readability, maintainability, and consistency.

### <samp>🤖 Generated by Copilot at 96c7c60</samp>

*  Import and use `unit` function from `@ant-design/cssinjs` to add units to numeric values in CSS objects ( [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-c24d0ab998b998a6a5cfd4d77148aad78198e1fe93395501020c8bbee7ebfc1cL1-R2), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-c24d0ab998b998a6a5cfd4d77148aad78198e1fe93395501020c8bbee7ebfc1cL26-R27), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L1-R2), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L30-R31), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L64-R65), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L191-R192), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L209-R210), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-76a3207c1405e5d3615abf7040b26c6e5366e5d9b863ba2f9a78194d727fe2d7L1-R2), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-76a3207c1405e5d3615abf7040b26c6e5366e5d9b863ba2f9a78194d727fe2d7L39-R40), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-76a3207c1405e5d3615abf7040b26c6e5366e5d9b863ba2f9a78194d727fe2d7L86-R91), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L1-R2), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L40-R41), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L52-R61), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L1-R2), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L28-R30), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L39-R41), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L59-R70), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L78-R85))
* Import and use `token.calc` function and `equal` method to perform arithmetic operations on token values in CSS objects ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L216-R223), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L265-R273), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-76a3207c1405e5d3615abf7040b26c6e5366e5d9b863ba2f9a78194d727fe2d7L55-R58), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L24-R25), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L52-R61), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931R22), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L49-R51), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L59-R70), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L78-R85))
* Import and use `useZIndex` hook from `_util` to control the z-index of the color picker component ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fR10))
* Import and use `useCSSVar` hook from `style` to register and apply CSS variables for the color picker component ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL33-R36))
* Replace `wrapSSR` function with `wrapCSSVar` function to wrap the color picker component with a CSS variable provider and a style tag ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL137-R139), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-be92dd92152a780003d48ddc95ec6716367d327f9f890f1221b84614aaa8168fL237-R239))
* Create a new file `cssVar.ts` in the `style` folder, which exports the `genCSSVarRegister` function from the `theme` folder with the `ColorPicker` component name and an empty object as arguments ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-830b273a91f41a04c542529c116795248963c781d95a0e2150e3135b0274b482R1-R3))
* Use `colorPickerPreviewSize` token to calculate the size of the color preview area in the color picker panel ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-1b0771e050d88aebe327dae413b7f31f0797bcee1515f3c6bb2cd890ba0929d0L265-R273))
* Use `fontHeightSM` token to set the height of the font size small in the collapse icon style ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931R22), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L28-R30))
* Use `getTransBg` function to set the background image of the alpha slider ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L52-R61))
* Adjust the spacing and alignment of various elements in the color picker panel, such as the select item, the hex input, the collapse icon, the presets list, and the transparent icon ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-76a3207c1405e5d3615abf7040b26c6e5366e5d9b863ba2f9a78194d727fe2d7L55-R58), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-76a3207c1405e5d3615abf7040b26c6e5366e5d9b863ba2f9a78194d727fe2d7L86-R91), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L28-R30), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L39-R41), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L49-R51), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L59-R70), [link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L78-R85))
* Set the border width and color of the handler element that moves along the color palette or slider ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L40-R41))
* Set the border radius of the slider and gradient elements that show the color spectrum or alpha channel ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L52-R61))
* Set the minimum height of the color palette area in the color picker panel ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-34b89f88ff38f571cda106c0e4980f0487a530c9c21623a0fd1695bb0b132a58L24-R25))
* Create a border effect around the selected preset color item ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L59-R70))
* Create a transparent icon effect on the preset color item that represents the transparent color ([link](https://github.com/ant-design/ant-design/pull/45868/files?diff=unified&w=0#diff-258bf706e0141cb724422da77ea2a1f43ecad3cdaeee2f0b86908eb12f1af931L78-R85))

### CODE REVIEW

File: components/color-picker/ColorPicker.tsx [hunk 1]
```diff
@@ -134,7 +135,8 @@ const ColorPicker: CompoundedComponent = (props) => {
 
   // ===================== Style =====================
   const mergedSize = useSize(customizeSize);
-  const [wrapSSR, hashId] = useStyle(prefixCls);
+  const [, hashId] = useStyle(prefixCls);
+  const wrapCSSVar = useCSSVar(prefixCls);
```
[the remark this answers is not in the document]
Comment (MadCcc) [reply components/color-picker/ColorPicker.tsx:139]: 可以改一下 ColorPicker 的 ComponentToken 测试下

[the remark this answers is not in the document]
Comment (RedJue) [reply components/color-picker/ColorPicker.tsx:139]: 1

File: components/color-picker/ColorPicker.tsx [hunk 2]
```diff
@@ -134,9 +136,11 @@ const ColorPicker: CompoundedComponent = (props) => {
 
   // ===================== Style =====================
   const mergedSize = useSize(customizeSize);
-  const [wrapSSR, hashId] = useStyle(prefixCls);
+  const [, hashId] = useStyle(prefixCls);
+  const rootCls = useCSSVarCls(prefixCls);
+  const wrapCSSVar = useCSSVar(rootCls);
   const rtlCls = { [`${prefixCls}-rtl`]: direction };
-  const mergeRootCls = classNames(rootClassName, rtlCls);
+  const mergeRootCls = classNames(rootClassName, rootCls, rtlCls);
```
Comment (MadCcc) [inline components/color-picker/ColorPicker.tsx:143]: 这样 popover 上是不是又少了 class

Comment (MadCcc) [reply components/color-picker/ColorPicker.tsx:143]: 这个 mergeRootCls 应该也要给到 Popover 吧

Comment (RedJue) [reply components/color-picker/ColorPicker.tsx:143]: 之前是没有给的，之前只给了这个
```tsx
const mergePopupCls = classNames(prefixCls, rtlCls);
```

**Comment by MadCcc** [comment]
lint 看一下

**Comment by RedJue** [comment]
> lint 看一下

好了

**Review by MadCcc — APPROVED** [review]

**Pull request merged**